src/wsdiscovery.ts

```diff
@@ -27,92 +36,173 @@ export default class WsDiscovery extends EventEmitter {
+                            }
+                        } else if (this.isLinkLocalAddr(networkInterface.address)) {
+                            linkLocalMaps.push({
+                                ip: networkInterface.address,
+                                interfaceName: k,
+                            });
                         }
                     }
                 });
             }
         }
 
-        return map;
+        return {
+            targetInterface,
+            linkLocalMaps,
+        };
+    }
+
+    watchNetworkInterfaces(): void {
+        if (this.opts.probeEntireNetwork || process.env.HUDDLY_WSDD_PROBE_ENTIRE_NETWORK) {
+            // No need to watch interfaces when socket is bound to all interfaces
+            return;
+        }
+
+        Logger.debug(`Setting up network interface watcher`, WsDiscovery.name);
+        const isNotInitialized = (map: InterfaceMap) => {
+            return !(map.interfaceName in this.socketConnections);
+        };
+
+        const checkForDisconnect = (maps: InterfaceMap[]) => {
+            const flattenedMaps = maps.map(m => m.interfaceName);
+            Object.keys(this.socketConnections).forEach(k => {
+                if (!flattenedMaps.includes(k)) {
+                    Logger.debug(
+                        `Network interface [${k}] not found. No longer doing discovery for Huddly Cameras on this interface.`,
+                        WsDiscovery.name
+                    );
+
+                    this.socketConnections[k].close();
+                    this.socketConnections[k] = undefined;
+                    delete this.socketConnections[k];
+                }
+            });
+        };
+
+        this.networkInterfacesWatcher = setInterval(() => {
+            const networkInterfaces = this.findNetworkInterfaces();
+            if (this.opts.targetInterfaceAddr || this.opts.targetInterfaceName) {
+                if (
+                    networkInterfaces.targetInterface.ip &&
+                    isNotInitialized(networkInterfaces.targetInterface)
+                ) {
+                    this.socketConnections[
+                        networkInterfaces.targetInterface.interfaceName
+                    ] = this.initSocketConnection(networkInterfaces.targetInterface);
+                }
+                checkForDisconnect([networkInterfaces.targetInterface]);
+                return;
+            }
+
+            if (networkInterfaces.linkLocalMaps) {
+                networkInterfaces.linkLocalMaps.forEach(map => {
+                    if (isNotInitialized(map)) {
+                        this.socketConnections[map.interfaceName] = this.initSocketConnection(map);
+                    }
+                });
+            }
+
+            checkForDisconnect(networkInterfaces.linkLocalMaps);
+        }, 1000);
+    }
+
+    bindSocket(map: InterfaceMap, socket: dgram.Socket): void {
+        if (!map.ip) socket.bind();
```

Comment:
You need to return inside the if. Otherwise it will try ti bind twice (line 178 and 180). Therefore you get the address in use.

Reply:
Good catch! Thank you for this :)